stem/model.py

```diff
@@ -882,6 +873,251 @@ def __initialise_fields(self):
+
+        Args:
+            - start_node_id (int): the node id to start searching the next node along the elements.
+            - remaining_element_ids (List[int]): the element ids that have not been followed yet.
+            - remaining_node_ids (List[int]): the node ids that have not been crossed yet.
+            - node_to_elements (Dict[int, List[int]]): mapping of node_ids to the element_ids which is connected to.
+            - line_elements (Dict[int, :class:`stem.mesh.Element`]): dictionary of line elements.
+            - target_node_ids (npty.NDArray[np.int64]): array of nodes to be searched for.
+
+        Raises:
+            - ValueError: if not all elements are line elements.
+            - ValueError: if there is a fork in the mesh, the algorithm cannot find the next node.
+            - ValueError: if number of interation in the while loop are exceeded and something went wrong in the
+                algorithm.
+
+        Returns:
+            - int: the node id which is connected to the start_node_id within the search space.
+
+        """
+
+        # check if all elements are line elements
+        for element in line_elements.values():
+            if element.element_type != "LINE_2N" and element.element_type != "LINE_3N":
+                raise ValueError("Not all elements are line elements.")
+
+        # initialise variables before loop
+        next_node = start_node_id
+
+        # start the search for the connected node
+        max_iterations = len(remaining_element_ids)
+        for i in range(max_iterations):
+
+            # find the element(s) connected to the node that have not yet been searched for.
+            elements_connected = [el for el in node_to_elements[next_node] if el in remaining_element_ids]
+
+            # check if there is a fork in the mesh, which is not allowed
+            if len(elements_connected) > 1:
+                raise ValueError(f"There is a fork in the mesh at elements: {elements_connected}, the next node along "
+                                 f"the line cannot be found.")
+
+            next_element_id = elements_connected[0]
+
+            # reduce search space for next iteration
+            remaining_element_ids.remove(next_element_id)
+
+            # find the node(s) connected to the element that have not yet been found yet.
+            for node_id in line_elements[next_element_id].node_ids:
+                if node_id in remaining_node_ids:
+                    next_node = node_id
+                    break
+
+            # reduce search space for next iteration
+            remaining_node_ids.remove(next_node)
+
+            # if the node is one of the nodes of interest, return them, otherwise continue.
+            if next_node in target_node_ids:
+                return next_node
+
+        raise ValueError("Next node along the line cannot be found. Maximum number of iterations exceeded.")
```

Comment:
it is not the maximum number of iteration. it is just that node is not part of the mesh.